docs/agent-reference/how-to-build-deviceupdate-agent-snap.md

```diff
@@ -63,23 +63,100 @@ tree squashfs-root
 
 To install the snap for testing (developer mode)
 
-```sh
+```shell
 sudo snap install --devmode ./deviceupdate-agent_0.1_amd64.snap 
 ```
 
 > NOTE | This will installs the snap content to /snap/<'snap name'>/ directory on the host device
 
+## Configure The Device Update Agent Snap
+
+Specify an agent settings, such as, manufacturer, model, agent.name, agent.connectionSource, etc., in `du-config.json`, which localted in `$SNAP_DATA/config` folder.
+
+```json
+{
+  "schemaVersion": "1.1",
+  "aduShellTrustedUsers": [
+    "snap_aziot_du",
+    "snap_aziot_do"
+  ],
+  "iotHubProtocol": "mqtt",
+  "manufacturer":"<Place your device info manufacturer here>",
+  "model": "<Place your device info model here>",
+  "agents": [
+    {
+      "name": "<Place your agent name here>",
+      "runas": "snap_aziot_du",
+      "connectionSource": {
+        "connectionType": "string",
+        "connectionData": "HostName=...HIDDEN..."
+      },
+      "manufacturer": "<Place your device property manufacturer here>",
+      "model": "<Place your device property model here>"
+    }
+  ]
+}
+```
+
+For example:
+- "manufacturer":"contoso"
+- "model":"vacuum-v1"
+- "agents[0].name":"main"
+- "agents[0].connectionSource.connectionType":"string"
+- "agents[0].connectionSource.connectionData":"**enter your device connection string here**"
+- "agents[0].manufacturer":"contoso"
+- "agents[0].model":"vacuum-v1"
+
+>TIP: To share file between host machine and the snap, you can use a ['home' interface](https://snapcraft.io/docs/home-interface).
+
+- To share 'home' directory with the snap, run:
+    ```shell
+    $ snap connect deviceupdate-agent:home :home
+    ```
+- Create du-config.json file in your home directory like so:
+    ```json
+    {
+        "schemaVersion": "1.1",
+        "aduShellTrustedUsers": [
+            "snap_aziot_du",
+            "snap_aziot_do"
+        ],
+        "iotHubProtocol": "mqtt",
+        "manufacturer": "contoso",
+        "model": "vacuum-v1",
+        "agents": [
+            {
+            "name": "main",
+            "runas": "snap_aziot_du",
+            "connectionSource": {
+                "connectionType": "string",
+                "connectionData": "HostName=...HIDDEN..."
+            },
+            "manufacturer": "contoso",
+            "model": "vacuum-v1"
+            }
+        ]
+    }
+    ```
+
+
 ## Run The Snap
 
-To run a shell in the confined environment
+To run a shell in the confined environment:
+
+```shell
+$ sudo snap run --shell deviceupdate-agent
+```
+
+Optionally, copy the prepared `du-config.json` from [host machine] home directory to `$SNAP_DATA/config` directory
 
-```sh
- snap run --shell deviceupdate-agent
+```shell
+$ cp <home-dir>/du-config.json $SNAP_DATA/config
```

Comment:
I used `vi`, but a little unsure about the `cp` here. Is it run inside the device-update snap shell? how do we specify the <home-dir>?

Reply:
Yes, this should be run inside the snap shell. Customer can choose their preferred method to author the du-config.json. As long as it is placed in the right location ($SNAP_DATA/config/du-config.json)